### ISSUE TITLE
Teams for Linux crashes on screen share

### ISSUE THREAD

**Issue by flavienlaurent** [IsmaelMartinez/teams-for-linux#669]
**Describe the bug**
When I'm sharing screens or windows in Teams, sometimes it works perfectly and sometime it crashes immediately with a segfault. Usually, from fresh start of Teams, I can screen share without any troubles and then when I try to share again after stopping the first one, it crashes.

**To Reproduce**
1. Make a call
2. Start sharing a screen or a window
3. Stop sharing
4. Start sharing again

**Expected behavior**
A stable screen sharing feature that works everytime.

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.1
 - Installation package deb
 - Windowing system: Wayland

**Additional context**
dmesg:
`traps: teams-for-linux[262696] general protection fault ip:7f0f55425014 sp:7fff822b1cd8 error:0 in libwayland-client.so.0.20.0[7f0f55423000+6000]`
process console:
`[270407:0120/162508.823088:ERROR:stun_port.cc(97)] Binding request timed out from 192.168.0.x:55715 (wlp59s0)
[1]    270309 segmentation fault (core dumped)  teams-for-linux`

Thanks

**Comment by jijojosephk** [comment]
I'm seeing no issue in X11. Possibly way land.

**Comment by jijojosephk** [comment]
We can try upgrading the dependencies.

**Comment by flavienlaurent** [comment]
I will try this today. Note that, it looks like it's often when I first screen share, then stop the sharing and then share again.

**Comment by jijojosephk** [comment]
@flavienlaurent can you run the app from terminal and share any logs if you find when this issue happens?

**Comment by flavienlaurent** [comment]
More than what I put in "Additional context" section?

**Comment by jubalfh** [comment]
note that hector martin found recently a bug in libwebrtc implementation that breaks screen sharing on wayland (cf. the [chromium bug](https://bugs.chromium.org/p/webrtc/issues/detail?id=15174) – might be the culprit here).

**Comment by jijojosephk** [comment]
Thanks @jubalfh for reporting that. Well, at least it has to reach the latest electron version. We're worried about the tray icon broken from electron version `22`.

**Comment by bernalvarela** [comment]
I'm having the same problem with the 1.2.2 teams-for-linux version.

The error i'm getting when i try to share is:

`
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[3778243:0712/094500.388974:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
Violación de segmento (`core' generado)
`

Regards!

**Comment by mcichocki** [comment]
@jijojosephk I also have problems with the stable operation of Teams-For-Linux on Ubuntu 22.04. At some point, it can disappoint the operating system and only helps restart the system. At the moment it is burdensome and the company's policy requires the use of teams for daily communication at work.

**Comment by wneuhaus** [comment]
I encounter a reproducible crash everytime when sharing my screen.

/opt/teams-for-linux$ ./teams-for-linux --appLogLevels=debug
[DEBUG] configPath:,/home/user/.config/teams-for-linux
[DEBUG] configFile:,[object Object]
[DEBUG] Fetching custom background configuration from 'http://localhost/config.json'
[DEBUG] Custom background service url is 'http://localhost'
[DEBUG] processArgs:,/opt/teams-for-linux/teams-for-linux,--appLogLevels=debug
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[93424:0718/091005.296714:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
[DEBUG] did-finish-load
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[93574:0718/091006.953973:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[DEBUG] did-finish-load
[DEBUG] Badge count set to '0'
[DEBUG] User status changed to '1'
[DEBUG] User status changed to '1'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[93574:0718/091016.596830:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[93574:0718/091023.855904:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] Power save is disabled using Electron API.
[DEBUG] User status changed to '5'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
Segmentation fault

**Desktop**:
OS: Debian GNU/Linux 12 (bookworm)
GNOME Version: 43.4
Windowing system: Wayland
Installation package: deb & flatpak

Please let me know how I can provide more information.

**Comment by anthonysmillie** [comment]
So my screen sharing had been working in Wayland until today. 

At the moment when I try to screen share I get an SSL error and the application crashes:

```
[58:0719/011628.667574:ERROR:ssl_client_socket_impl.cc(992)] handshake failed; returned -1, SSL error code 1, net_error -3
[58:0719/011628.667821:ERROR:ssl_client_socket_impl.cc(992)] handshake failed; returned -1, SSL error code 1, net_error -3
```

Full log:

```
[2:0719/011605.571166:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
LaunchProcess: failed to execvp:
xdg-settings
[2:0719/011605.783159:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0719/011605.783190:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[WARN] Unsupported language code 'en-ZA' for spellchecker
[2:0719/011605.824576:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0719/011605.824606:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0719/011605.824625:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[ERROR] connect ECONNREFUSED ::1:80
[2:0719/011608.535455:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0719/011608.535481:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0719/011608.535511:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0719/011608.535530:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0719/011608.535548:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[52:0719/011608.542168:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[58:0719/011628.667574:ERROR:ssl_client_socket_impl.cc(992)] handshake failed; returned -1, SSL error code 1, net_error -3
[58:0719/011628.667821:ERROR:ssl_client_socket_impl.cc(992)] handshake failed; returned -1, SSL error code 1, net_error -3
```

**Comment by anthonysmillie** [comment]
Ignore my last update comment here, managed to fix the issue by updating.

**Comment by wneuhaus** [comment]
> I encounter a reproducible crash everytime when sharing my screen.
> 
> /opt/teams-for-linux$ ./teams-for-linux --appLogLevels=debug [DEBUG] configPath:,/home/user/.config/teams-for-linux [DEBUG] configFile:,[object Object] [DEBUG] Fetching custom background configuration from 'http://localhost/config.json' [DEBUG] Custom background service url is 'http://localhost' [DEBUG] processArgs:,/opt/teams-for-linux/teams-for-linux,--appLogLevels=debug [DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active' [93424:0718/091005.296714:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented [DEBUG] did-finish-load [DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active' [93574:0718/091006.953973:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times! [DEBUG] did-finish-load [DEBUG] Badge count set to '0' [DEBUG] User status changed to '1' [DEBUG] User status changed to '1' [DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active' [93574:0718/091016.596830:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times! [93574:0718/091023.855904:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times! [DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active' [DEBUG] Power save is disabled using Electron API. [DEBUG] User status changed to '5' [DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active' [DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active' Segmentation fault
> 
> **Desktop**: OS: Debian GNU/Linux 12 (bookworm) GNOME Version: 43.4 Windowing system: Wayland Installation package: deb & flatpak
> 
> Please let me know how I can provide more information.

This is no further an issue for me as I have moved on to Linux Mint where screen sharing works flawlessly.

**Comment by jijojosephk** [comment]
Can you try the latest version? `1.2.9`

**Comment by enzy** [comment]
The v1.2.9 works for me (or at least it doesn't crash and pretends it's sharing).

**Comment by jijojosephk** [comment]
Is it working for everyone?

**Comment by andreaippo** [comment]
For me it wasn't crashing on Wayland, rather stop refreshing screen contents and only show my mouse move around, but the other person would see it moving over an old/frozen in time "canvas"/window/app.

I'm glad to report that following some testing this morning with version 1.2.9, it's been flawless.

Test only lasted a bit under 10 minutes, but with a previous version just a couple of weeks ago, I would have faced that issue already. I also tried to stop and restart screen sharing during the same call, and it's behaved as expected.

Thanks to the devs! I stumbled upon this thread because I was looking for a fix for MY issue, but since I'm not the OP, I think he has the last word on whether HIS issue is fixed.
